base/repl/REPLCompletions.jl

```diff
@@ -462,9 +462,9 @@ function dict_identifier_key(str,tag)
         isa(obj, Array) && return (nothing, nothing, nothing)
     end
     begin_of_key = first(search(str, r"\S", nextind(str, end_of_identifier) + 1)) # 1 for [
-    begin_of_key==0 && return (true, nothing, nothing)
+    begin_of_key==0 && return (nothing, nothing, nothing)
```

Comment:
Do the return statement on line 496, do anything with this change? This is also ok, as it gives the other types of completion a chance.

Reply:
Good question, probably not. But I have several questions regarding some of the code here, so I went with the minimal change. For example, it seems that line 462 (special casing Array objects) is no longer necessary. Probably processing the key (i.e. line 464) should be done after the check for being a dictionary on line 467, etc.